rr_rviz_plugins/include/rr_rviz_plugins/ResetPanel.h

```diff
@@ -0,0 +1,30 @@
+#ifndef PROJECT_RESETPANEL_H
+#define PROJECT_RESETPANEL_H
+
+#include <ros/ros.h>
+#include <rviz/panel.h>
+#include <QPushButton>
+#include <std_msgs/Empty.h>
+
+/*
+ * All of our panels need to be under the rr_rviz_plugins namespace.
+ */
+namespace rr_rviz_plugins {
+
+class ResetPanel : public rviz::Panel {
+Q_OBJECT
+public:
+    ResetPanel(QWidget *parent = 0);
+
+protected:
+    ros::NodeHandle nh;
```

Comment:
Consider matching the ExamplePanel a little more closely. You don't actually need to save _nh_ or _reset_btn_ beyond the scope of the constructor, so they don't need to be member variables of the class. Also, while default initializers (as used here for _reset_pub_ are awesome, it's generally cleaner to only use the for trivially construct-able types.

Reply:
Strangely, moving any of the three statements (declaring the button, declaring the NodeHandle, and advertising) out of the header and into the .cpp causes the button to stop working. I've tried all combinations and the only thing that works is keeping it as is. This site enforces the fact for creating QPushButtons: https://wiki.qt.io/How_to_Use_QPushButton